internal/validator/validator.go

```diff
@@ -15,10 +17,57 @@ type OutpointData struct {
 	Satoshis     int64
 }
 
-type Validator interface {
-	// ValidateEFTransaction Please note that bt.Tx should have all the fields of each input populated.
-	ValidateEFTransaction(tx *bt.Tx, skipFeeValidation bool, skipScriptValidation bool) error
-	ValidateBeef(beef *beef.BEEF, skipFeeValidation bool, skipScriptValidation bool) (*bt.Tx, error)
+type FeeValidation byte
+
+const (
+	NoneFeeValidation FeeValidation = iota
+	StandardFeeValidation
+)
+
+type ScriptValidation byte
+
+const (
+	NoneScriptValidation ScriptValidation = iota
+	StandardScriptValidation
+)
+
+type DefaultValidator interface {
+	ValidateTransaction(ctx context.Context, tx *bt.Tx, feeValidation FeeValidation, scriptValidation ScriptValidation) error
 	IsExtended(tx *bt.Tx) bool
-	IsBeef(txHex []byte) bool
+}
+
+type BeefValidator interface {
+	ValidateTransaction(ctx context.Context, beef *beef.BEEF, feeValidation FeeValidation, scriptValidation ScriptValidation) (*bt.Tx, error)
+}
+
+type HexFormat byte
+
+const (
+	RawHex HexFormat = iota
+	EfHex
+	BeefHex
+)
+
+func GetHexFormat(hex []byte) HexFormat {
+	if beef.CheckBeefFormat(hex) {
+		return BeefHex
+	}
+
+	if isEf(hex) {
+		return EfHex
+	}
+
+	return RawHex
+}
+
+func isEf(hex []byte) bool {
```

Comment:
Now we have 2 functions to check if a tx is EF. 
The other one is this one: https://github.com/bitcoin-sv/arc/blob/60a3c1b587647645aa8e601a05fc0fd8457bb481/internal/validator/default/default_validator.go#L59:L71

Wouldn't it be better to have just one? 🤔

Reply:
The one you mentioned checks if `bt.Tx` structure contains all the necessary information to be treated as extended. This one checks the hex format sent by the client.
So on this stage we need both